src/librustc_typeck/astconv.rs

```diff
@@ -947,14 +961,6 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx>+'o {
         )
     }
 
-    /// Transform a `PolyTraitRef` into a `PolyExistentialTraitRef` by
-    /// removing the dummy `Self` type (`TRAIT_OBJECT_DUMMY_SELF`).
-    fn trait_ref_to_existential(&self, trait_ref: ty::TraitRef<'tcx>)
```

Comment:
Instead of removing this function, you could extend it to include the entire erasure procedure